### PR TITLE
Fix #17583 - Set production fee address for Gemini

### PR DESCRIPTION
**Pull request by jumde**
Resolves https://github.com/brave/brave-browser/issues/17583

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Tip a gemini verified creator
2. Verify 5% of the tip is deposited to the fees-address

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -2368,7 +2368,7 @@ void RewardsServiceImpl::HandleFlags(const std::string& options) {
       std::string lower = base::ToLowerASCII(value);
 
       if (lower == "true" || lower == "1") {
-        environment = ledger::type::Environment::STAGING;
```
Comment (emerick) [inline components/brave_rewards/browser/rewards_service_impl.cc:2371]: Left over test code?

**Review by emerick — APPROVED** [review]
LGTM

**Comment by Miyayes** [comment]
Verified using:

Nightly 1.30.40

```
Brave | 1.30.40 Chromium: 93.0.4577.51 (Official Build) nightly (x86_64)
-- | --
Revision | 762d21050e2da59930c784c09b134d0b0b148188-refs/branch-heads/4577@{#915}
OS | macOS Version 11.4 (Build 20F71)

```

<details>

1. Made tip of 1.00 BAT

2. After 0.95 BAT portion completed 

3. 0.05 BAT (fee portion) initiated to the new fee address in this PR. 

image

I made several other tips too just to for 0.25 BAT and 0.5 BAT (to other sites), and also seeing the fee portions:

image

image
</details>

**Comment by jumde** [comment]
Fees transaction on Gemini - 

Screen Shot 2021-08-23 at 10 43 03 AM

**Pull request merged**